lib/src/fnc/vector.rs

```diff
@@ -1,8 +1,49 @@
 use crate::err::Error;
-use crate::fnc::util::math::dotproduct::DotProduct;
-use crate::fnc::util::math::magnitude::Magnitude;
+use crate::fnc::util::math::vector::{
+	Add, Angle, CrossProduct, Divide, DotProduct, Magnitude, Multiply, Normalize, Project, Subtract,
+};
 use crate::sql::{Number, Value};
 
+pub fn add((a, b): (Vec<Number>, Vec<Number>)) -> Result<Value, Error> {
+	match a.add(&b) {
+		None => Err(Error::InvalidArguments {
+			name: String::from("vector::add"),
+			message: String::from("The two vectors must be of the same length."),
```

Comment:
"dimension" is probably more appropriate than "length" here

Reply:
Agreed. `length` for arrays, `dimension` for vectors